src/gatsby/src/pages/find-a-park.js

```diff
@@ -705,236 +786,126 @@ export default function FindAPark({ location, data }) {
-                                        control={
-                                          <Checkbox
-                                            checked={
-                                              selectedActivities.filter(
-                                                act => act.value === a.value
-                                              ).length === 1
-                                                ? true
-                                                : false
-                                            }
-                                            onChange={event => {
-                                              handleActivityCheck(a, event)
-                                            }}
-                                            name={a.label}
-                                          />
-                                        }
-                                        label={a.label}
-                                        className={
-                                          selectedActivities.filter(
-                                            act => act.value === a.value
-                                          ).length === 1
-                                            ? "text-light-blue no-wrap"
-                                            : "no-wrap"
-                                        }
-                                      />
-                                    )
-                                  })}
-                                </FormGroup>
-                                <Link
-                                  className="ml-auto pointer p20"
-                                  onClick={() => {
-                                    handleActivitiesLengthChange([])
-                                  }}
-                                  onKeyUp={(e) => {
-                                    if (e.key === " " || e.key === "Enter") {
-                                      handleActivitiesLengthChange([])
-                                    }
-                                  }}
-                                  tabIndex="0"
-                                  role="link"
-                                  underline="hover"
-                                >
-                                  {showMoreActivities ? (
-                                    <div style={{ color: `#2464A4` }}>
-                                      Show all {activityItems.length}
-                                      <ExpandMore fontSize="small" />
-                                    </div>
-                                  ) : (
-                                    <div style={{ color: `#2464A4` }}>
-                                      Show less
-                                      <ExpandLess fontSize="small" />
-                                    </div>
-                                  )}
-                                </Link>
+                            {showMoreActivities ? (
```

Comment:
The text updates too soon after you click the link. Can we show a spinner or some sort of progress indicator until the expand/collapse is complete?